metadata/src/utils/retain.rs

```diff
@@ -208,50 +287,24 @@ pub fn retain_metadata<F, G>(
     F: FnMut(&str) -> bool,
     G: FnMut(&str) -> bool,
 {
-    let mut type_ids = HashSet::new();
+    // Types specifically referenced inside pallets that we keep
+    let mut retained_set = TypeSet::new();
 
-    // There are special outer enum types that point to all pallets types (call, error, event) by default.
-    // This brings in a significant chunk of types. We trim this down to only include variants
-    // for the pallets we're retaining, to avoid this.
-    retain_pallets_in_runtime_outer_types(metadata, &mut pallets_filter);
-
-    // Collect the stripped outer enums.
-    collect_outer_enums(&metadata.outer_enums, &mut type_ids);
-
-    // Filter our pallet list to only those pallets we want to keep. Keep hold of all
-    // type IDs in the pallets we're keeping. Retain all, if no filter specified.
-    metadata.pallets.retain(|pallet| {
+    for pallet in metadata.pallets.values() {
         let should_retain = pallets_filter(&pallet.name);
         if should_retain {
-            collect_pallet_types(pallet, &mut type_ids);
+            retained_set.collect_pallet_types(pallet, metadata);
         }
-        should_retain
-    });
-
-    // We index pallets by their u8 index for easy access. Rebuild this index.
-    metadata.pallets_by_index = metadata
-        .pallets
-        .values()
-        .iter()
-        .enumerate()
-        .map(|(pos, p)| (p.index, pos))
-        .collect();
-
-    // Keep the extrinsic stuff referenced in our metadata.
-    collect_extrinsic_types(&metadata.extrinsic, &mut type_ids);
-
-    // Keep the "runtime" type ID, since it's referenced in our metadata.
-    type_ids.insert(metadata.runtime_ty);
+    }
 
-    // Keep only the runtime API types that the filter allows for. Keep hold of all
-    // type IDs in the runtime apis we're keeping. Retain all, if no filter specified.
-    metadata.apis.retain(|api| {
+    // all types that we intend to keep
+    let mut type_set = retained_set.clone();
```

Comment:
Why is `retained_set` cloned here and not used anymore? For readability?
It doesn't seem necessarily to me

Reply:
I think this is just a hangover from the prev version! If so, I'd vote to just call it `type_set` throughout :)

Reply:
removed retained_set